server/mailrepository/mailrepository-blob/src/main/scala/org/apache/james/mailrepository/blob/BlobMailRepository.scala

```diff
@@ -138,12 +135,12 @@ object BlobMailRepository {
   }
 }
 
-class BlobMailRepository(val blobStore: BlobStoreDAO,
+class BlobMailRepository @Inject() (val blobStore: BlobStoreDAO,
                          val blobIdFactory: BlobId.Factory,
-                         val mimeMessageStore: Store[MimeMessage, MimeMessagePartsId]
-                        ) extends MailRepository {
-  private val bucketName: BucketName = BucketName.of("mailMetadata")
-  private val mimeMessageBucketName: BucketName = BucketName.of("mimeMessageData")
+                         val mimeMessageStore: Store[MimeMessage, MimeMessagePartsId],
+                         val url: MailRepositoryUrl) extends MailRepository {
```

Comment:
as I said in my email, this is something we were going to be working on in our next session but if you need it faster than that, then sure :D

Reply:
Well it is more that without this this implem is trivially broken...

That's also a good opportunity to harden the mail repository contract.

Reply:
Improving the contract does sound good.

We will resolve the conflicts and finalize the fix when we get there as this won't be enough since we probably also want to get a different mimeMessageStore...